wordsFrequency/main.cpp

```diff
@@ -0,0 +1,129 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <utility>
+#include <vector>
+#include <cctype>
+#include <list>
+#include <algorithm>
+#include <limits>
+#include <optional>
+#include <set>
+#include <cassert>
+
+void addToFirstTen(std::vector<std::pair<std::string, int>>& words, const std::string& word, int frequency) {
+    bool inserted = false;
+
+    for(ulong i = 0; i < words.size(); i++) {
+        if(frequency > words[i].second)
+        {
+            auto it = words.begin() + i;
+            words.insert(it, std::pair(word, frequency));
```

Comment:
Secondo me qui è quasi più semplice chiamare direttamente `words.cbegin() + i` al posto di usare la variabile locale `it`.